test/posix_time/testduration.cpp

```diff
@@ -17,28 +17,34 @@
 
 
 int
-main() 
+main()
 {
   using namespace boost::posix_time;
 
-  
+
 //   std::cout << "Default limits: SECOND " << std::endl;
   {
     time_duration td;
     check("default construction -- 0 ticks", td.ticks() == 0);
 //     check("default construction -- 0 secs", td.seconds() == 0);
-//     check("default construction -- 0 min", td.minutes() == 0); 
+//     check("default construction -- 0 min", td.minutes() == 0);
   }
-  
+
   // construct from components
   time_duration td1(1,25,0);
   time_duration td3(td1.hours(),td1.minutes(),td1.seconds());
   check("total up elements", td1 == td3);
   td1 = -td1; // td1 == "-1:25:00"
+  check("invert_sign",td3 == td1.invert_sign());
+  check("abs",td3 == td1.abs());
+  check("is_positive",td3.is_positive());
```

Comment:
And an is_positive for the false case.  You can see the unit test coverage is flagging missed paths.